cmd/chihaya/main.go

```diff
@@ -65,9 +65,17 @@ func (r *Run) Start(ps storage.PeerStore) error {
 	if err != nil {
 		return errors.New("failed to validate hook config: " + err.Error())
 	}
+	preHookNames := make([]string, 0)
+	for _, hookCfg := range cfg.PreHooks {
+		preHookNames = append(preHookNames, hookCfg.Name)
+	}
```

Comment:
This could be made a method of the `cfg.Prehook` and `cfg.Posthook` and then the log line will look like:

```go
log.WithFields(log.Fields{
  "prehooks": prehooks.Names(),
  "posthooks": posthooks.Names(),
}).Info("starting middleware")
```